k8s/core/configmap/configmap_lock_v2_test.go

```diff
@@ -226,7 +226,7 @@ func TestCMLockRefreshV2(t *testing.T) {
 			err = cm.LockWithKey(id1, key1)
 			require.NoError(t, err, "Unexpected error in LockWithKey(id1,key1)")
 
-			require.True(t, reentrantCheck.CompareAndSwap(false, true), "Reentrant lock detected")
+			require.False(t, reentrantCheck.CompareAndSwap(false, true), "Reentrant lock detected")
```

Comment:
We wanted to keep the lock as non-reentrant so that the locking mechanism can be simple and avoid potential complicated locking scenarios

Reply:
Seems like my change is breaking this non-rentrant logic